client/signup/steps/design-picker/index.jsx

```diff
@@ -113,11 +113,11 @@ export default function DesignPickerStep( props ) {
 		setSelectedDesign( designs.find( ( { theme } ) => theme === props.stepSectionName ) );
 	}, [ designs, props.stepSectionName, setSelectedDesign ] );
 
-	const [ selectedCategory, setSelectedCategory ] = useCategorySelection(
-		designs,
-		props.showDesignPickerCategoriesAllFilter,
-		props.signupDependencies.intent === 'write' ? 'blog' : null
-	);
+	const categorization = useCategorization( designs, {
```

Comment:
I think this component might not need to know `useCategorization` hook. We should give the relative props to `<DesignPicker />` and it would call this hook. In further, if `showDesignPickerCategories` is false, this hook won't trigger if possible. But it's okay for this PR and we can create another PR for further improvement.

Reply:
Not using the hook when `showDesignPickerCategories` is false would be nice.

I don't think calling `useCategorization()` in `<DesignPicker>` works because when the user previews a design the `<DesignPicker>` is unmounted and the hook will be torn down. When we go back to the gallery from the preview we have to recreate the `<DesignPicker>`, which will create a fresh hook and the currently selected category will have been forgotten.

Unless there's something I'm missing?

> We should give the relative props to `<DesignPicker />`

Maybe I misunderstood this. Do you mean you prefer passing individual props (e.g. current selection, sort function, etc.) to the component instead of one single "categorization" object?

Reply:
Oh yes, you're right. I forgot the `<DesignPicker />` will umount when the user previews a design.

> Maybe I misunderstood this. Do you mean you prefer passing individual props (e.g. current selection, sort function, etc.) to the component instead of one single "categorization" object?

I mean we just need to pass the options of `useCategorization` to the `DesignPicker` and then we don't need to use that hook here. But as we need to keep the current selection after going back from the preview. So, you can ignore this idea now.